pkg/operator/controller/ingress/deployment.go

```diff
@@ -1253,6 +1253,7 @@ func hashableDeployment(deployment *appsv1.Deployment, onlyTemplate, defaultIngr
 			LivenessProbe:   hashableProbe(container.LivenessProbe, defaultIngressController),
 			ReadinessProbe:  hashableProbe(container.ReadinessProbe, defaultIngressController),
 			StartupProbe:    hashableProbe(container.StartupProbe, defaultIngressController),
+			SecurityContext: container.SecurityContext,
```

Comment:
You need also need to do the update in `deploymentConfigChanged`:
```diff
--- a/pkg/operator/controller/ingress/deployment.go
+++ b/pkg/operator/controller/ingress/deployment.go
@@ -1467,6 +1467,7 @@ func deploymentConfigChanged(current, expected *appsv1.Deployment, defaultIngres
 	}
 	updated.Spec.Template.Spec.Volumes = volumes
 	updated.Spec.Template.Spec.NodeSelector = expected.Spec.Template.Spec.NodeSelector
+	updated.Spec.Template.Spec.Containers[0].SecurityContext = expected.Spec.Template.Spec.Containers[0].SecurityContext
 	updated.Spec.Template.Spec.Containers[0].Env = expected.Spec.Template.Spec.Containers[0].Env
 	updated.Spec.Template.Spec.Containers[0].Image = expected.Spec.Template.Spec.Containers[0].Image
 	copyProbe(expected.Spec.Template.Spec.Containers[0].LivenessProbe, updated.Spec.Template.Spec.Containers[0].LivenessProbe, defaultIngressController)
```
Please add a test case for this:
```diff
--- a/pkg/operator/controller/ingress/deployment_test.go
+++ b/pkg/operator/controller/ingress/deployment_test.go
@@ -1317,6 +1317,17 @@ func TestDeploymentConfigChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if the container security context changes",
+			mutate: func(deployment *appsv1.Deployment) {
+				v := true
+				sc := &corev1.SecurityContext{
+					AllowPrivilegeEscalation: &v,
+				}
+				deployment.Spec.Template.Spec.Containers[0].SecurityContext = sc
+			},
+			expect: true,
+		},
 	}
 
 	for _, tc := range testCases {
```